addons/mail/static/tests/message/message_tests.js

```diff
@@ -672,6 +673,24 @@ QUnit.test("should not be able to reply to temporary/transient messages", async
     await contains(".o-mail-Message [title='Reply']", { count: 0 });
 });
 
+QUnit.test("squashed transient message should not have date in the sidebar", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "Channel 1", channel_type: "chat" });
+    const { openDiscuss } = await start();
+    openDiscuss(channelId);
+    await insertText(".o-mail-Composer-input", "/who");
+    await click(".o-mail-Composer-send:enabled");
+    await contains(".o-mail-Message");
+
+    await insertText(".o-mail-Composer-input", "/who", { replace: true });
+    await click(".o-mail-Composer-send:enabled");
+    await click(".o-mail-Message.o-squashed");
+    await nextTick();
+    await contains(".o-mail-Message.o-squashed .o-mail-Message-sidebar .o-mail-Message-date", {
+        count: 0,
+    });
+});
```

Comment:
In general negative testing are poor, because the sidebar could be shown and it's just the way that is asserted that is incorrect and the test pass.

To assert a negative, we should assert a positive targeting a similar thing. I'd create 2 messages in setup of test that should squash one of them and assert there's a sidebar with date.

Also for asserting sidebar with date, you can simplify the selector with presence of a visible date on a given message. so if test is run with mocked date at `12:00` assertion should tell there's this `12:00` displayed (or not) on this message. This is a better assertion because it matches what the user sees, rather than assertions on presence of classnames for which the user doesn't see at all.

Reply:
The issue with asserting positive is that the condition which is responsible (`t-elif="!message.is_transient"`) would always be true if `is_transient` is not there or is false. 
The purpose of the test case is to make sure that Transient messages never have date in their sidebar.
There was another issue because of this which was recently closed where on hovering on transient messages was causing flickers as the date was being displayed in 12 hour format which would have never happened if the date was not being shown there.

Reply:
Got your point :+1: 
Please take a look at the changes.